api/napi/index.ts

```diff
@@ -144,4 +145,75 @@ export class ArrayModel<T> implements Model<T> {
     entries(): IterableIterator<[number, T]> {
         return this.a.entries()
     }
+}
+
+export class Component {
+    private instance: ComponentInstance;
+
+    /**
+    * @hidden
+    */
+    constructor(instance: ComponentInstance) {
+        this.instance = instance;
+    }
+
+    run() {
+        this.instance.run();
+    }
+
+    show() {
+        this.instance.window().show();
+    }
+
+    hide() {
+        this.instance.window().hide();
+    }
+
+    get window(): Window {
+        return this.instance.window();
+    }
+}
+
+/**
+ * @hidden
+ */
+interface Callback {
+    (): any;
+    setHandler(cb: any): void;
+}
+
+export function create(path: string): Component | null {
+    let compiler = new ComponentCompiler;
+    let definition = compiler.buildFromPath(path);
+
+    if (definition != null) {
```

Comment:
This deeply nested code makes me realise that the ComponentCompiler/ComponentDefinition API needs perhaps a change to properly accommodate error handling.

Our current interpreter API has `build_from_path` return an `Option` and if it returns None, then you can call `diagnostics()`. I wonder why we did it this way instead of returning a `Result`. @ogoffart do you remember by chance?

For the JavaScript API, I'm thinking that perhaps it's idiomatic to either return a non-null object (so we don't need these checks) or we throw an exception (in case of an error).

Reply:
I looked a bit at the history.
At first it used to return a `Result<ComponentDefinition, ComponentLoadError>`
But then, I changed it in https://github.com/slint-ui/slint/commit/5c79f2dbe1e0bfe2b6f81f1967bfc54926fc6938 to be a  `-> (Option<Self>, Vec<Diagnostic>)`  because i needed to access the diagnostics also when it succeeds  (eg: warnings).
Later when polishing the API, you converted to the statefull API in https://github.com/slint-ui/slint/commit/4a9cd954b865e01391919ca4cc0a865e6095d495#diff-25d7ad16ce622c20d7211a39c429dcdc6bc6e6ca52e3b39aeef584f01b560a0bL397